worlds/seaofthieves/Locations/LocationCollection.py

```diff
@@ -0,0 +1,193 @@
+import random
+
+from .Voyager.IslandVisited import VoyageIslandVisited
+from .Voyager.VoyageQuestGh import VoyageQuestGh
+from .Voyager.VoyageQuestOos import VoyageQuestOos
+from .Voyager.VoyageQuestMa import VoyageQuestMa
+from .Voyager.VoyageQuestRor import VoyageQuestRor
+from .Voyager.VoyageQuestAthena import VoyageQuestAthena
+from .Rouge.RogueQuestAll import RogueQuestAll
+from .Feared.FearedQuestSeaForts import FearedQuestSeaForts
+from .Menu.QuestMenu import MenuQuestAll
+from .Locations import SOTLocation, LocDetails
+from .Hunter.ProvisionsCooked import BurntAboard, CookedAboard, Total
+from .Hunter.ProvisionsEaten import EatenAboard
+from .Seals import Seals
+from .Goldseaker import TreasuresSold, Chests
+from ..Configurations import SotOptionsDerived
+from .Servant import Servant
+from .Guardian import Guardian
+from .IllFated import IllFated
+from .Feared import CannonsFired
+from ..Items.Items import Item
+from ..Regions.RegionConnectionRules import RegionDiver
+from ..Regions.ConnectionDetails import ConnectionDetails
+from .Voyager.CaptainShipSpotted import CaptainShipSpotted
+from .Voyager import DaysAtSea
+from .Voyager import NauticalMilesSailed
+from .Voyager import Rowboats
+from .Voyager import Shipwrecks
+from .Voyager import TallTales
+from .Shops import Shops
+from .Shop import ShopLocation, Balance
+import typing
+from .Shop import ShopWarehouse, ShopLocationList, ShopLocation
+
+
+class LocationDetailsCollection:
+
+    SHOP_DESCRIPTOR = "shop"
+    def __init__(self):
+        LocDetails.resetSeedId()
+        self.options: SotOptionsDerived.SotOptionsDerived = SotOptionsDerived.SotOptionsDerived()
+
+        # this maps Settings Class -> [locName -> LocDets]
+        self.checkTypeToLocDetail: typing.Dict[str, typing.Dict[str, LocDetails]] = {}
+
+        self.hintWebLocations: [LocDetails] = []
+
+        self.regionDiver: RegionDiver | None = None
+
+        # TODO this may cause bugs.. we should move the prices out of the shops here somehow to decouple this?
+        self.random: random.Random | None = None
```

Comment:
You can't `type` like this in python 3.8, which is still supported. Hopefully 3.8 support will be dropped before this is merged.